### PR TITLE
🏗Components folder restructure

### PR DESCRIPTION
**Pull request by josemigallas**
We're locating all components under the same roof (folder `components`) and using the same folder structure as the routes, leaving a folder `shared` for the generic ones. This is how it's going to look:
```js
components/
  pages/                                       // -> Components used only in one page
    login/
      LoginPage.tsx                            // -> Main, routed component
      LoginForm.tsx                            // -> Children components
    audience/                                  // -> Section
      overview/  
      analytics/
      accounts/                                // -> Subsection
        listing/                               // -> Page
          DeveloperAccounts.tsx                // -> Main, routed component
          DeveloperAccountsTable.tsx           // -> Children components used in the main view
          ...
  shared/                                      // -> Shared components
    empty-states/
    data-list/
    utils/
    ...
```

### CODE REVIEW

**Comment by didierofrivia** [comment]
There's still `src/components/developer-accounts` directory, that IMHO, it should be inside `src/components/pages/developer-accunts`

File: portafly/src/components/shared/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// export * from './AppNavItem'
+export * from './data-list'
+export * from './EmptyState'
+// export * from './LazyRoute'
+// export * from './PrivateRoute'
+export * from './Loading'
+// export * from './NotFound'
+// export * from './SwitchWith404'
+// export * from './util'
+// export * from './Root'
```
Comment (didierofrivia) [inline portafly/src/components/shared/index.tsx:14]: Any reason of keeping this commented lines?

File: portafly/src/tests/components/shared/data-list/modals/DataListModal.test.tsx [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+
 import { render } from 'tests/custom-render'
 import { fireEvent } from '@testing-library/react'
+import { DataListModal } from 'components'
 
 it('shows up to 5 items and a button to expand the list', () => {
   const items = new Array(10).fill('test').map((admin, i) => admin + i)
   const { getByText, getAllByText } = render(<DataListModal
-    actions={[]}
+    submitButton={<input key="submit" />}
```
Comment (didierofrivia) [inline portafly/src/tests/components/shared/data-list/modals/DataListModal.test.tsx:10]: Maybe this change alongside the ones in `ChangeStateModal.test.tsx` should be in a separated commit (if not a different PR)

File: portafly/src/components/shared/index.tsx (hunk 1 above)
Comment (josemigallas) [reply portafly/src/components/shared/index.tsx:14]: 🤯

File: portafly/src/tests/components/shared/data-list/modals/DataListModal.test.tsx (hunk 2 above)
Comment (josemigallas) [reply portafly/src/tests/components/shared/data-list/modals/DataListModal.test.tsx:10]: Totally agree, I'll remove it

**Pull request merged**